src/main/java/com/okta/developer/web/rest/AccountResource.java

```diff
@@ -46,7 +48,15 @@ public AccountResource(UserService userService) {
     @GetMapping("/account")
     @SuppressWarnings("unchecked")
     public AdminUserDTO getAccount(Principal principal) {
-        Optional<User> user = userService.getUserWithAuthoritiesByLogin(principal.getName());
+        // FIXME: really ugly line, could cause NPE and class cast exceptions
+        String email =
+            ((OAuth2AuthenticationToken) principal).getPrincipal()
+                .getAttributes()
+                .getOrDefault("email", "")
+                .toString()
+                .toLowerCase(Locale.ROOT);
```

Comment:
error-prone line 😢

Reply:
`SecurityUtils.getCurrentUserLogin()` might be easier.